File: server/backend/app/services/mail_record_service.py
"""邮件发送记录服务 - CRUD。敏感字段（gpt_password/email_password）后端已加密存储。"""
from sqlalchemy import select, delete
from sqlalchemy.ext.asyncio import AsyncSession

from ..models import MailRecord
from ..utils.helpers import now_utc


async def add_mail_record(db: AsyncSession, record: dict) -> int:
    """新增发送记录。敏感字段写入时由模型层加密（encrypt_field 在 router 处理）。"""
    m = MailRecord(**record)
    db.add(m)
    await db.flush()
    return m.id


async def list_mail_records(db: AsyncSession) -> list[MailRecord]:
    """查询所有记录（按发送时间倒序）。敏感字段读取时解密。"""
    stmt = select(MailRecord).order_by(MailRecord.sent_at.desc())
    return list((await db.execute(stmt)).scalars().all())


async def delete_mail_record(db: AsyncSession, rid: int) -> None:
    """删除单条记录。不存在时抛 ValueError（与 expense_service.delete_expense 一致）。"""
    existing = await db.get(MailRecord, rid)
    if existing is None:
        raise ValueError("邮件记录不存在")
    await db.delete(existing)


async def clear_mail_records(db: AsyncSession) -> None:
    await db.execute(delete(MailRecord))

File: src/services/mailRecordService.ts
// 邮件发送记录服务 - 数据层迁移最后一批：走后端 API。
// 敏感字段（gpt_password/email_password）由后端加密存储、API 返回明文。
// 前端不再做加解密。
import type { MailRecord } from '@/types';
import { apiClient } from './apiClient';

// 日期归一化
function normalizeRecord(r: any): MailRecord {
  return { ...r, sent_at: new Date(r.sent_at) };
}

// 新增发送记录（后端加密敏感字段）
export async function addMailRecord(record: Omit<MailRecord, 'id'>): Promise<number> {
  const r = await apiClient.post<{ id: number }>('/api/mail-records', record);
  return r.id;
}

// 查询所有发送记录（后端解密敏感字段，返回明文）
export async function listMailRecords(): Promise<MailRecord[]> {
  const list = await apiClient.get<any[]>('/api/mail-records');
  return list.map(normalizeRecord);
}

// 删除单条记录
export async function deleteMailRecord(id: number): Promise<void> {
  await apiClient.delete(`/api/mail-records/${id}`);
}

// 清空所有记录
export async function clearMailRecords(): Promise<void> {
  await apiClient.delete('/api/mail-records');
}
